src/threads.lisp

```diff
@@ -16,16 +16,18 @@
   ((threads :accessor pool-threads :initform nil)
    (tasks :accessor pool-tasks :initform nil)))
 
-(defclass soft-thread-pool (thread-pool)
-  ((threads :accessor pool-threads :initform nil)
-   (free-thread-counter :accessor free-thread-counter :initform 0)
-   (soft-limit :accessor pool-soft-limit :initform 1000) ; this seems like a sane-ish default
-   (lock :reader pool-lock :initform (bt:make-lock "thread pool lock"))
+(defclass leader-follower-pool (thread-pool)
+  ((lock :reader pool-lock :initform (bt:make-lock "thread pool lock"))
+   (leader-notifier :reader pool-leader-notifier :initform
+             (bt:make-condition-variable :name "pool leader notifier"))
+   (free-thread-counter :accessor free-thread-counter :initform 0)))
+
```

Comment:
needs more documentation

Reply:
despite technically being internals, due to chanl's goal of being the only threading dependency of libraries using it, the various tools for handling various encountered situations caused by the thread pools' logic must be documented.